src/tests/ftest/util/ior_utils.py

```diff
@@ -1,110 +1,267 @@
+        #         raise IorFailed("Missing required {} param".format(name))
+
+        if self.api.value == "MPIIO":
+            env = {
+                "CRT_ATTACH_INFO_PATH": attach_info_path,
+                "DAOS_POOL": str(self.daos_pool),
+                "MPI_LIB": "",
+                "DAOS_SVCL": str(self.daos_svcl),
+                "DAOS_SINGLETON_CLI": 1,
+                "FI_PSM2_DISCONNECT": 1,
+            }
+            export_cmd = [
+                "export {}={}".format(key, val) for key, val in env.items()]
+            mpirun_cmd = [
+                os.path.join(build_paths["OMPI_PREFIX"], "bin/mpirun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+            ]
+            command = " ".join(mpirun_cmd + [self.__str__()])
+            command = "; ".join(export_cmd + command)
+
+        elif self.api.value == "DAOS":
+            orterun_cmd = [
+                os.path.join(build_paths["OMPI_PREFIX"], "bin/orterun"),
+                "-np {}".format(processes),
+                "--hostfile {}".format(hostfile),
+                "--map-by node",
+                "-x DAOS_SINGLETON_CLI=1",
+                "-x CRT_ATTACH_INFO_PATH={}".format(attach_info_path),
+            ]
+            command = " ".join(orterun_cmd + [self.__str__()])
+
+        else:
+            raise IorFailed("Unsupported IOR API: {}".format(self.api))
+
+        return command
+
+    def run(self, basepath, processes, hostfile, display_output=True):
+        """Run the IOR command.
+
+        Args:
+            basepath (str): DAOS base path
+            processes (int): number of host processes
+            hostfile (str): file defining host names and slots
+            display_output (bool, optional): print IOR output to the console.
+                Defaults to True.
+
+        Raises:
+            IorFailed: if an error occured runnig the IOR command
+
+        """
+        command = self.get_launch_command(basepath, processes, hostfile)
+        if display_output:
+            print("<IOR CMD>: {}".format(command))
+
+        # Run IOR
+        try:
+            process = subprocess.Popen(
+                command, stdout=subprocess.PIPE, shell=True)
+            while True:
+                output = process.stdout.readline()
+                if output == "" and process.poll() is not None:
+                    break
+                if output and display_output:
+                    print(output.strip())
+            if process.poll() != 0:
+                raise IorFailed(
+                    "IOR Run process Failed with non zero exit code:{}".format(
+                        process.poll()))
+
+        except (OSError, ValueError) as error:
+            print("<IorRunFailed> Exception occurred: {0}".format(str(error)))
+            raise IorFailed("IOR Run process Failed: {}".format(error))
 
 
 def run_ior_daos(client_file, ior_flags, iteration, block_size, transfer_size,
```

Comment:
This function should make use of the new classes:

```
def run_ior_daos(client_file, ior_flags, iteration, block_size, transfer_size,
                 pool_uuid, svc_list, object_class, basepath, client_processes,
                 cont_uuid="`uuidgen`", seg_count=1, chunk_size=1048576,
                 display_output=True):
    """Run IOR test.

    Args:
        client_file (str): file holding client hostname and slots
        ior_flags (str): all ior specific flags
        iteration (int): number of iterations for ior run
        block_size (int): contiguous bytes to write per task
        transfer_size (int): size of transfer in bytes
        pool_uuid (str): DAOS pool uuid
        svc_list (str): DAOS pool svcl
        object_class (str): DAOS object class designation
        basepath (str): DAOS base path
        client_processes (int): number of client processes
        cont_uuid (str, optional): container uuid. Defaults to "`uuidgen`".
        seg_count (int, optional): segment count. Defaults to 1.
        chunk_size (int, optional): chunk size in bytes. Defaults to 1048576.
        display_output (bool, optional): print IOR output. Defaults to True.
    """
    ior_cmd = IorCommand()
    ior_cmd.api.value = "DAOS"
    ior_cmd.flags.value = ior_flags
    ior_cmd.repetitions.value = iteration
    ior_cmd.block_size.value = block_size
    ior_cmd.transfer_size.value = transfer_size
    ior_cmd.daos_pool.value = pool_uuid
    ior_cmd.daos_svcl.value = svc_list
    ior_cmd.daos_oclass.value = object_class
    ior_cmd.daos_cont.value = cont_uuid
    ior_cmd.segment_count.value = seg_count
    ior_cmd.daos_chunk.value = chunk_size

    ior_cmd.run(
        basepath, client_processes, client_file, display_output)
```

Reply:
I am using slurm for the launcher so I am only generating the ior cmdline an then using slurm (srun) to launch